[2025-04-25T06:52:30.881+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-04-25T06:52:30.901+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: stock_kafka_pipeline.spark_job scheduled__2025-04-25T06:40:00+00:00 [queued]>
[2025-04-25T06:52:30.907+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: stock_kafka_pipeline.spark_job scheduled__2025-04-25T06:40:00+00:00 [queued]>
[2025-04-25T06:52:30.907+0000] {taskinstance.py:2306} INFO - Starting attempt 2 of 2
[2025-04-25T06:52:30.919+0000] {taskinstance.py:2330} INFO - Executing <Task(SparkSubmitOperator): spark_job> on 2025-04-25 06:40:00+00:00
[2025-04-25T06:52:30.925+0000] {standard_task_runner.py:63} INFO - Started process 1016 to run task
[2025-04-25T06:52:30.927+0000] {standard_task_runner.py:90} INFO - Running: ['airflow', 'tasks', 'run', 'stock_kafka_pipeline', 'spark_job', 'scheduled__2025-04-25T06:40:00+00:00', '--job-id', '367', '--raw', '--subdir', 'DAGS_FOLDER/test.py', '--cfg-path', '/tmp/tmpm5dfwfzp']
[2025-04-25T06:52:30.929+0000] {standard_task_runner.py:91} INFO - Job 367: Subtask spark_job
[2025-04-25T06:52:30.962+0000] {task_command.py:426} INFO - Running <TaskInstance: stock_kafka_pipeline.spark_job scheduled__2025-04-25T06:40:00+00:00 [running]> on host 9489df9b8294
[2025-04-25T06:52:31.035+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='stock_kafka_pipeline' AIRFLOW_CTX_TASK_ID='spark_job' AIRFLOW_CTX_EXECUTION_DATE='2025-04-25T06:40:00+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-04-25T06:40:00+00:00'
[2025-04-25T06:52:31.036+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-04-25T06:52:31.052+0000] {base.py:84} INFO - Using connection ID 'spark_default' for task execution.
[2025-04-25T06:52:31.053+0000] {spark_submit.py:328} ERROR - Failed to decode base64 keytab: Incorrect padding
[2025-04-25T06:52:31.053+0000] {spark_submit.py:305} INFO - Could not load connection string spark_default, defaulting to spark://172.21.0.8:7077:7077
[2025-04-25T06:52:31.054+0000] {spark_submit.py:474} INFO - Spark-Submit cmd: spark-submit --master spark://172.21.0.8:7077:7077 --conf spark.driver.memory=2g --conf spark.executor.memory=4g --principal *** --name arrow-spark --queue default --deploy-mode client Source/test.py
[2025-04-25T06:52:31.067+0000] {spark_submit.py:644} INFO - /home/***/.local/bin/find-spark-home: line 40: /usr/local/bin/python3.10: No such file or directory
[2025-04-25T06:52:31.068+0000] {spark_submit.py:644} INFO - /home/***/.local/bin/spark-submit: line 27: /bin/spark-class: No such file or directory
[2025-04-25T06:52:31.068+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-04-25T06:52:31.077+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 197, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 566, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://172.21.0.8:7077:7077 --conf spark.driver.memory=2g --conf spark.executor.memory=4g --principal *** --name arrow-spark --queue default --deploy-mode client Source/test.py. Error code is: 127.
[2025-04-25T06:52:31.080+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=stock_kafka_pipeline, task_id=spark_job, run_id=scheduled__2025-04-25T06:40:00+00:00, execution_date=20250425T064000, start_date=20250425T065230, end_date=20250425T065231
[2025-04-25T06:52:31.222+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 367 for task spark_job (Cannot execute: spark-submit --master spark://172.21.0.8:7077:7077 --conf spark.driver.memory=2g --conf spark.executor.memory=4g --principal *** --name arrow-spark --queue default --deploy-mode client Source/test.py. Error code is: 127.; 1016)
[2025-04-25T06:52:31.260+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-04-25T06:52:31.275+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-04-25T06:52:31.277+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
